src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -0,0 +1,330 @@
+#
+# SMB.CONF(5)		The configuration file for the Samba suite 
+# $FreeBSD$
+#
+<%
+        import os
+        import sys
+        sys.path.extend([
+            '/usr/local/www',
+            '/usr/local/www/freenasUI'
+        ])
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from freenasUI.common.freenassysctl import freenas_sysctl as fs
+        parsed_conf = {}
+
+        def get_db_config():
+            conf = {}
+
+            conf['cifs'] = middleware.call_sync('datastore.query', 'services.cifs', None, {'get': True})
+            conf['ad'] = middleware.call_sync('datastore.query', 'directoryservice.activedirectory', None, {'get': True})
+            conf['ldap'] = middleware.call_sync('datastore.query', 'directoryservice.ldap', None, {'get': True})
+            conf['NIS'] = middleware.call_sync('datastore.query', 'directoryservice.nis', None, {'get': True})
+            conf['gc'] = middleware.call_sync('datastore.query', 'network.globalconfiguration', None, {'get': True})
+            conf['role'] = "standalone"
+
+            conf['truenas_conf'] = {"is_truenas_ha": False, "failover_status": "DEFAULT", "smb_ha_mode": None}
+            if not middleware.call_sync('notifier.is_freenas') and middleware.call_sync('notifier.failover_licensed'):
+                conf['truenas_conf']['is_truenas_ha'] = True
+                conf['truenas_conf']['failover_status'] = middleware.call_sync('notifier.failover_status')
+                if conf['cifs']['cifs_srv_netbiosname'] == conf['cifs']['cifs_srv_netbiosname_b']:
+                    conf['truenas_conf']['smb_ha_mode'] = "unified"
+
+            if conf['ad']['ad_enable']:
+                conf['role'] = "ad_member"
+            elif conf['ldap']['ldap_enable'] and conf['ldap']['ldap_has_samba_schema']:
+                conf['role'] = "ldap_member"
+            elif middleware.call_sync('datastore.query', 'services.services', [('srv_service','=','domaincontroller')], {'get' : True})['srv_enable']:
+                conf['role'] = "domain_controller"
+                conf['DC'] = middleware.call_sync('datastore.query', 'services.DomainController', None, {'get': True})
+
+            return conf 
+
+        def get_sysctl_config():
+            conf = {}
+            conf['min_protocol'] = fs().services.smb.config.server_min_protocol 
+            conf['max_protocol'] = fs().services.smb.config.server_max_protocol
+            #conf['multichannel'] = fs().services.smb.config.server_multi_channel
+
+            return conf
+
+        def get_cifs_homedir():
+            cifs_homedir = "/home"
+                
+            shares = middleware.call_sync('datastore.query', 'sharing.CIFS_Share')
```

Comment:
`sharing.smb.query, [["home", "=", True]]`

would probably simplify this quite a bit.